it/uk/gov/hmrc/transitmovementsrouter/connectors/PersistenceConnectorSpec.scala

```diff
@@ -139,8 +139,62 @@ class PersistenceConnectorSpec
       stub(OK)
 
       val failingSource = Source.single(ByteString.fromString("<abc>asdadsadads")).via(new EISMessageTransformersImpl().unwrap)
+      implicit val hc: HeaderCarrier =
+        HeaderCarrier(extraHeaders = Seq("X-Message-Type" -> MessageType.DeclarationAmendment.code, HeaderNames.CONTENT_TYPE -> MimeTypes.XML))
+      whenReady(connector.postBody(movementId, messageId, DeclarationAmendment, failingSource).value) {
+        res =>
+          res mustBe a[Left[Unexpected, _]]
+          res.left.toOption.get.asInstanceOf[Unexpected].thr.isDefined
+      }
+    }
+
+  }
+
+  "post for Large Message" should {
+    "return messageId when post is successful" in {
+      val body           = Json.obj("messageId" -> messageId.value).toString()
+      val objectStoreURI = s"common-transit-convention-traders/movements/$movementId/x-conversion-id.xml"
+      stub(OK, Some(body))
+      implicit val hc: HeaderCarrier = HeaderCarrier(extraHeaders =
+        Seq("X-Message-Type" -> MessageType.DeclarationAmendment.code, "X-Object-Store-Uri" -> ObjectStoreURI(objectStoreURI).value)
+      )
```

Comment:
And here too. In fact, this is worse because I would NOT expect to have the `X-Object-Store-Uri` or `X-Message-Type` header in the header carrier at all (in the previous case, we'd only have the `Content-Type`). Additionally, because you're relying on the lack of a content type at the target, then we really need to be checking for that too.